src/components/shared/Widget/PriceChart.vue

```diff
@@ -791,25 +796,42 @@ export default class PriceChartWidget extends Mixins(
     const lastTimestamp = last(this.dataset)?.timestamp ?? Date.now();
 
     this.priceUpdateRequestId = requestId;
-
     await this.withApi(async () => {
       try {
-        const snapshots = await Promise.all(addresses.map((address) => this.fetchData(address)));
+        const snapshots: Snapshot[] = this.loadDataWhenReverse
+          ? addresses.map(
+              (address) => this?.snapshotBuffer?.[address] || { nodes: [], hasNextPage: false, endCursor: undefined }
```

Comment:
Could you please move this code to `fetchData ` method? So that this method contains logic for working with buffers and loading data (in the absence of a buffer)